src/wireless.c

```diff
@@ -407,23 +407,128 @@ ni_wireless_scan_add_bss(ni_wireless_scan_t *scan, ni_wireless_bss_t *bss)
 	*next = bss;
 }
 
-static uint32_t
-ni_wireless_frequency_to_channel(uint16_t frequency)
+unsigned int
+ni_wireless_frequency_to_channel(unsigned int frequency)
 {
-	if (frequency > 5000){
+	if (frequency >= 5950) {
+		return (frequency - 5950) / 5;
+
+	} else if (frequency > 5000) {
 		return (frequency - 5000) / 5;
 
-	} else if (frequency >= 4915){
+	} else if (frequency >= 4915) {
 		return (frequency - 4915) / 5 + 183;
 
-	} else if (frequency == 2484){
+	} else if (frequency == 2484) {
 		return 14;
 
 	} else {
 		return (frequency - 2407) / 5;
 	}
 }
 
+ni_bool_t
+ni_wireless_band2freq_list(ni_uint_array_t *list, const char *value)
+{
+	static const unsigned int chan_2_4ghz[] = {
```

Comment:
This aren't channels but frequencies:

```suggestion
	static const unsigned int freq_set_2_4ghz[] = {
```